br/pkg/restore/import.go

```diff
@@ -240,12 +240,53 @@ func (ic *importClient) GetImportClient(
 		grpc.WithConnectParams(grpc.ConnectParams{Backoff: bfConf}),
 		grpc.WithKeepaliveParams(ic.keepaliveConf),
 	)
-	if err != nil {
-		return nil, errors.Trace(err)
+	return conn, errors.Trace(err)
+}
+
+func (ic *importClient) GetImportClient(
+	ctx context.Context,
+	storeID uint64,
+) (import_sstpb.ImportSSTClient, error) {
+	ic.mu.Lock()
+	defer ic.mu.Unlock()
+	conn, ok := ic.conns[storeID]
+	if ok {
+		return import_sstpb.NewImportSSTClient(conn), nil
 	}
-	client = import_sstpb.NewImportSSTClient(conn)
-	ic.clients[storeID] = client
-	return client, errors.Trace(err)
+	conn, err := ic.createGrpcConn(ctx, storeID)
+	ic.conns[storeID] = conn
+	return import_sstpb.NewImportSSTClient(conn), errors.Trace(err)
+}
+
+func (ic *importClient) GetIngestClient(
```

Comment:
Maybe extract the duplicated code to a private function like:
```go
func (ic *importClient) cachedConnectionFrom(
	ctx context.Context,
	storeID uint64,
	caches map[uint64]*grpc.ClientConn,
) (import_sstpb.ImportSSTClient, error) {
	ic.mu.Lock()
	defer ic.mu.Unlock()
	conn, ok := caches[storeID]
	if ok {
		return import_sstpb.NewImportSSTClient(conn), nil
	}
	conn, err := ic.createGrpcConn(ctx, storeID)
	caches[storeID] = conn
	return import_sstpb.NewImportSSTClient(conn), errors.Trace(err)
}
```
and delegate the two functions `Get(Ingest|Import)Client` to it.